File: fix_client_metadata.py
#!/usr/bin/env python3
"""
Script to fix 'use client' directive conflicts with metadata exports
"""
import os
import re
import glob

def fix_client_metadata(file_path):
    """Fix 'use client' directive conflicts with metadata exports"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has both 'use client' and metadata export
        if "'use client'" in content and 'export const metadata' in content:
            # Remove the 'use client' directive
            content = re.sub(r"'use client';\n?", '', content)
            
            # Write the fixed content back
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            
            print(f"Fixed client/metadata conflict in: {file_path}")
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix all client/metadata conflicts"""
    # Find all TypeScript/JavaScript files in app directory
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts'
    ]
    
    fixed_count = 0
    total_files = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            # Skip node_modules and other directories
            if any(skip in file_path for skip in ['node_modules', '.git', 'dist', '.next']):
                continue
                
            total_files += 1
            if fix_client_metadata(file_path):
                fixed_count += 1
    
    print(f"\nFixed client/metadata conflicts in {fixed_count} out of {total_files} files")

if __name__ == "__main__":
    main()

File: fix-client-metadata.js
import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';

// Find all page.tsx files
function findPageFiles(dir) {
  const files = [];
  const items = fs.readdirSync(dir);
  
  for (const item of items) {
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    
    if (stat.isDirectory()) {
      files.push(...findPageFiles(fullPath));
    } else if (item === 'page.tsx') {
      files.push(fullPath);
    }
  }
  
  return files;
}

// Fix client directive and metadata conflicts
function fixClientMetadata(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Check if file has 'use client' directive
    const hasUseClient = content.includes("'use client'");
    
    if (hasUseClient) {
      // Remove metadata export if 'use client' is present
      content = content.replace(/export const metadata = \{[\s\S]*?\};/g, '');
      
      // Clean up extra semicolons and empty lines
      content = content.replace(/;\s*;/g, ';');
      content = content.replace(/\n\s*\n\s*\n/g, '\n\n');
    }
    
    // If we made changes, write the file back
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Main function
function main() {
  console.log('Starting to fix client directive and metadata conflicts...');
  
  const appDir = path.join(process.cwd(), 'app');
  const pageFiles = findPageFiles(appDir);
  
  console.log(`Found ${pageFiles.length} page files`);
  
  let fixedCount = 0;
  
  for (const filePath of pageFiles) {
    if (fixClientMetadata(filePath)) {
      fixedCount++;
    }
  }
  
  console.log(`Fixed ${fixedCount} files`);
  
  // Run build to verify
  try {
    console.log('Running build...');
    execSync('npm run build', { stdio: 'inherit' });
    console.log('Build passed!');
  } catch (error) {
    console.error('Build failed:', error.message);
  }
}

main();